syncano_cli/sync/project.py

```diff
@@ -48,13 +48,35 @@ def update_from_instance(self, instance, all=False, classes=None,
                              scripts=None):
         """Updates project data from instances"""
         LOG.info("Pulling instance data from syncano")
+        prev_classes = self.classes
         classes = classes or self.classes.keys()
         scripts = scripts or set(s['label'] for s in self.scripts)
         if all:
             classes = None
             scripts = None
         self.classes = pull_classes(instance, classes)
         self.scripts = pull_scripts(instance, scripts)
+
+        def cmpDicts(d1, d2):
+            """
+            Compare two dicts returning added, removed, modified, same
+            """
+            d1_keys = set(d1.keys())
+            if not d2:
+                return None, d1, None, None
```

Comment:
In other place you return 3 sets and iterable, and here you return None, d1, None, None. Be consistent in returning objects with the same type. So If you really insist on this `if` it should be `return set(), set(), set(), set()`